napari/util/misc.py

```diff
@@ -269,18 +269,19 @@ def calc_data_range(data):
     if np.prod(data.shape) > 1e6:
         # If data is very large take the average of the top, bottom, and
         # middle slices
-        top_plane_idx = np.zeros(data.ndim - 2).astype(int)
-        bottom_plane_idx = np.subtract(data.shape[:-2], 1).astype(int)
-        middle_plane_idx = np.round(np.divide(bottom_plane_idx, 2)).astype(int)
-        top_plane = np.asarray(data[tuple(top_plane_idx)])
-        bottom_plane = np.asarray(data[tuple(bottom_plane_idx)])
-        middle_plane = np.asarray(data[tuple(middle_plane_idx)])
-        reduced_data = np.array([top_plane, bottom_plane, middle_plane])
+        bottom_plane_idx = (0,) * (data.ndim - 2)
+        middle_plane_idx = tuple(s // 2 for s in data.shape[:-2])
+        top_plane_idx = tuple(s - 1 for s in data.shape[:-2])
+        idxs = [bottom_plane_idx, middle_plane_idx, top_plane_idx]
+        reduced_data = [
+            [np.max(data[idx]) for idx in idxs],
+            [np.min(data[idx]) for idx in idxs],
+        ]
     else:
         reduced_data = data
 
-    min_val = reduced_data.min()
-    max_val = reduced_data.max()
+    min_val = np.min(reduced_data)
```

Comment:
I think @jni this change has caused some of our tests to break when I run pytest locally - but I'm not quite sure why they went through on this PR - for example

```
napari/tests/test_numpy_like.py:31:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
napari/components/viewer_model.py:477: in add_image
    visible=visible,
napari/layers/image/image.py:191: in __init__
    self._contrast_limits_range = calc_data_range(input_data)
napari/util/misc.py:286: in calc_data_range
    min_val = np.min(reduced_data)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

args = (dask.array<random_sample, shape=(10, 15, 6, 16), dtype=float64, chunksize=(10, 15, 6, 16)>,), kwargs = {}
relevant_args = (dask.array<random_sample, shape=(10, 15, 6, 16), dtype=float64, chunksize=(10, 15, 6, 16)>, None)

>   ???
E   TypeError: no implementation found for 'numpy.amin' on types that implement __array_function__: [<class 'dask.array.core.Array'>]
```

Reply:
ah you are well acquainted with this issue!! looks like it is fixed in the new dask release - https://github.com/dask/dask/issues/5031

Side comment I think we should add info on all our named dependencies to our about menu! it's just so convenient

Reply:
all tests now pass for me